src/client/localplayer.cpp

```diff
@@ -392,15 +392,20 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 			// Move player to the maximal height when falling or when
 			// the ledge is climbed on the next step.
 
-			// Smoothen the movement (based on 'position.Y = bmax.Y')
-			position.Y += y_diff * dtime * 22.0f + BS * 0.01f;
-			position.Y = std::min(position.Y, bmax.Y);
-			m_speed.Y = 0.0f;
+			// Check if there is space for sneak glitch
+			v3f pos(position.X, position.Y + y_diff * dtime * 22.0f + BS * 0.01f, position.Z);
+			if (!collisionInCollision(env, m_client, m_collisionbox, pos)) {
+				// Smoothen the movement (based on 'position.Y = bmax.Y')
+				position.Y += y_diff * dtime * 22.0f + BS * 0.01f;
+				position.Y = std::min(position.Y, bmax.Y);
+				m_speed.Y = 0.0f;
+			}
```

Comment:
this is calculating the new Y position twice

also, couldn't you use `wouldCollideWithCeiling` instead of adding a new function to `collision.cpp`?

Reply:
I have tried to use wouldCollideWithCeiling to replace
https://github.com/minetest/minetest/blob/d802230718221c9d7587bbcc91f6fcc2b9c292c4/src/collision.cpp#L677-L696.
But it does not work as expected.